scripts/api.py

```diff
@@ -0,0 +1,148 @@
+# non-public API (don't host publicly) 
+# (will only be on with --api starting option)
+
+import numpy as np
+from fastapi import FastAPI, Body
+from fastapi.exceptions import HTTPException
+from PIL import Image
+
+import gradio as gr
+
+from modules.api.models import *
+from modules.api import api
+from modules.shared import opts
+
+from src.core import core_generation_funnel
+from src.common_ui import main_ui_panel
+from src.misc import SCRIPT_VERSION
+
+def encode_to_base64(image):
+    if type(image) is str:
+        return image
+    elif type(image) is Image.Image:
+        return api.encode_pil_to_base64(image)
+    elif type(image) is np.ndarray:
+        return encode_np_to_base64(image)
+    else:
+        return ""
+
+def encode_np_to_base64(image):
+    pil = Image.fromarray(image)
+    return api.encode_pil_to_base64(pil)
+
+def to_base64_PIL(encoding: str):
+    return Image.fromarray(np.array(api.decode_base64_to_image(encoding)).astype('uint8'))
+
+#TODO: is this slow?
+def get_defaults():
+    default_gradio = main_ui_panel(True).internal
+    defaults = {}
+    for key, value in default_gradio.items(): 
+        defaults[key]= value.value
+    return defaults
+
+def depth_api(_: gr.Blocks, app: FastAPI):
+    @app.get("/depth/version")
+    async def version():
+        return {"version": SCRIPT_VERSION}
+
+    @app.get("/depth/get_options")
+    async def get_options():
+        default_input =  get_defaults()
+        return {"settings": sorted(list(default_input.internal.keys()))}
+    
+    #This will be the stable basic api
+    @app.post("/depth/process")
+    async def process(
+        depth_input_images: List[str] = Body([], title='Input Images'),
+        compute_device:str = Body("GPU", title='CPU or GPU', options="'GPU', 'CPU'"), 
+        model_type:str = Body('zoedepth_n (indoor)', title='depth model', options="'res101', 'dpt_beit_large_512 (midas 3.1)', 'dpt_beit_large_384 (midas 3.1)', 'dpt_large_384 (midas 3.0)', 'dpt_hybrid_384 (midas 3.0)', 'midas_v21', 'midas_v21_small', 'zoedepth_n (indoor)', 'zoedepth_k (outdoor)', 'zoedepth_nk'"),
+        net_width:int = Body(512, title="net width"), 
+        net_height:int = Body(512, title="net height"), 
+        match_size:bool =  Body(True, title="match original image size"), 
+        boost:bool = Body(False, title="use boost algorithm"), 
+        invert_depth:bool = Body(False, title="invert depthmap")
+    ):
+        default_inputs = get_defaults()
+        override = {
+            # TODO: These indexing aren't soo nice
+            'compute_device': compute_device, 
+            'model_type': ['res101', 'dpt_beit_large_512 (midas 3.1)',
```

Comment:
Model types should be reworked, maybe as an enum, where every element of the Enum (model) is assigned a string. In future we might add a thing called "display name", so this rework should be extendable in this way. Passing bare strings is suboptimal - there be typos! Also, integers are bad - we sometimes want to insert models inside the list not at the end. Maybe we will even remove support for some models, eventually, or at least hide from the UI, by default.